sql/core/src/test/resources/sql-tests/inputs/subquery/exists-subquery/exists-joins-and-set-ops.sql

```diff
@@ -0,0 +1,206 @@
+                        Count(*) 
+                 FROM   emp 
+                        JOIN dept 
+                          ON dept.dept_id = emp.dept_id 
+                 WHERE  bonus.emp_name = emp.emp_name 
+                 GROUP  BY dept.dept_id, 
+                           emp.emp_name 
+                 ORDER  BY emp.emp_name);
+
+-- Aggregations in outer and subquery + join in subquery
+-- TC.01.08
+SELECT emp_name, 
+       Sum(bonus_amt) 
+FROM   bonus 
+WHERE  EXISTS (SELECT emp_name, 
+                        Max(salary) 
+                 FROM   emp 
+                        JOIN dept 
+                          ON dept.dept_id = emp.dept_id 
+                 WHERE  bonus.emp_name = emp.emp_name 
+                 GROUP  BY emp_name 
+                 HAVING Count(*) > 1 
+                 ORDER  BY emp_name)
+GROUP  BY emp_name; 
+
+-- TC.01.09
+SELECT emp_name, 
+       Sum(bonus_amt) 
+FROM   bonus 
+WHERE  NOT EXISTS (SELECT emp_name, 
+                          Max(salary) 
+                   FROM   emp 
+                          JOIN dept 
+                            ON dept.dept_id = emp.dept_id 
+                   WHERE  bonus.emp_name = emp.emp_name 
+                   GROUP  BY emp_name 
+                   HAVING Count(*) > 1 
+                   ORDER  BY emp_name) 
+GROUP  BY emp_name;
+
+-- Set operations along with EXISTS subquery
+-- union
+-- TC.02.01 
+SELECT * 
+FROM   emp 
+WHERE  EXISTS (SELECT * 
+               FROM   dept 
+               WHERE  dept_id < 30 
+               UNION 
+               SELECT * 
+               FROM   dept 
+               WHERE  dept_id >= 30 
+                      AND dept_id <= 50); 
+
+-- intersect 
+-- TC.02.02 
+SELECT * 
+FROM   emp 
+WHERE  EXISTS (SELECT * 
+                 FROM   dept 
+                 WHERE  dept_id < 30 
+                 INTERSECT 
+                 SELECT * 
+                 FROM   dept 
+                 WHERE  dept_id >= 30 
+                        AND dept_id <= 50);
+
+-- intersect + not exists 
+-- TC.02.03                
+SELECT * 
+FROM   emp 
+WHERE  NOT EXISTS (SELECT * 
+                     FROM   dept 
+                     WHERE  dept_id < 30 
+                     INTERSECT 
+                     SELECT * 
+                     FROM   dept 
+                     WHERE  dept_id >= 30 
+                            AND dept_id <= 50); 
+
+-- Union in outer query and except,intersect in subqueries. 
+-- TC.02.04       
+SELECT * 
+FROM   emp 
+WHERE  EXISTS (SELECT * 
+                 FROM   dept 
+                 EXCEPT 
+                 SELECT * 
+                 FROM   dept 
+                 WHERE  dept_id > 50)
+UNION ALL 
+SELECT * 
+FROM   emp 
+WHERE  EXISTS (SELECT * 
+                 FROM   dept 
+                 WHERE  dept_id < 30 
+                 INTERSECT 
+                 SELECT * 
+                 FROM   dept 
+                 WHERE  dept_id >= 30 
+                        AND dept_id <= 50);
+
```

Comment:
@dongjoon-hyun Thanks !! I have added the UNION test case. I have also uploaded the updated db2 result file.